src/graph/Node.cpp

```diff
@@ -113,7 +118,40 @@ void Node::enqueueExec()
 		auto msg = fmt::format("{}: attempted to call enqueueExec() despite !isValid()", getName());
 		throw std::logic_error(msg);
 	}
-	this->enqueueExecImpl(getGraph()->getStream()->get());
-	CHECK_CUDA(cudaEventRecord(execCompleted, getGraph()->getStream()->get()));
+	this->enqueueExecImpl(getGraphRunCtx()->getStream()->get());
+	CHECK_CUDA(cudaEventRecord(execCompleted, getGraphRunCtx()->getStream()->get()));
 }
 
+std::set<Node::Ptr> Node::getConnectedNodes()
+{
+	if (hasGraphRunCtx()) {
+		return getGraphRunCtx()->getNodes();
+	}
+	std::set<Ptr> visited = {};
+	std::function<void(Ptr)> dfsRec = [&](Ptr current) {
```

Comment:
So, this case, asume graph which has no graphRunCtx, or what?

Reply:
Precisely, we do early return when graphRunCtx is available and compute connected nodes otherwise. This code was actually moved from `Graph::findConnectedNodes(Node::Ptr)`